File: app/src/main/java/com/commonpepper/photosen/ui/viewmodels/SearchFragmentViewModelFactory.java
package com.commonpepper.photosen.ui.viewmodels;

import com.commonpepper.photosen.network.datasource.SearchListDataSourceFactory;
import com.commonpepper.photosen.model.Photo;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

public class SearchFragmentViewModelFactory implements ViewModelProvider.Factory {

    private String query;
    private String tags;
    private String sort;

    public SearchFragmentViewModelFactory(String query, String tags, String sort) {
        this.query = query;
        this.tags = tags;
        this.sort = sort;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        return (T) new SearchListFragmentViewModel();
    }

    public class SearchListFragmentViewModel extends AbstractListFragmentViewModel<Photo> {

        @Override
        public SearchListDataSourceFactory createDataSourceFactory() {
            return new SearchListDataSourceFactory(query, tags, sort);
        }
    }

}

File: app/src/main/java/com/commonpepper/photosen/ui/viewmodels/SearchFragmentViewModelFactory.kt
package com.commonpepper.photosen.ui.viewmodels

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider.Factory
import com.commonpepper.photosen.model.Photo
import com.commonpepper.photosen.network.datasource.SearchListDataSourceFactory

class SearchFragmentViewModelFactory(private val query: String, private val tags: String, private val sort: String) : Factory {
    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        return SearchListFragmentViewModel() as T
    }

    inner class SearchListFragmentViewModel : AbstractListFragmentViewModel<Photo?>() {
        override fun createDataSourceFactory(): SearchListDataSourceFactory {
            return SearchListDataSourceFactory(query, tags, sort)
        }
    }

}